src/insights_client/tests/test_metrics.py

```diff
@@ -34,27 +35,92 @@ def factory(config):
-        )
+%s
+%s
+""" % tuple(
+            "%s = %s" % (key, config[key]) if key in config else ""
+            for key in ("hostname", "port", "proxy_hostname", "proxy_port", "proxy_user", "proxy_password", "repo_ca_cert", "consumerCertDir"))
         return _tempfile(config)
 
     return factory
 
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_missing_server_section(proxy_settings, config_file_factory):
+    '''
+    Verify that when the "server" section is missing, RHSM configuration is ignored
+    and insights-client configuration is preferred
+    '''
+    config_file = config_file_factory("username=testuser\npassword=testpass\nauthmethod=BASIC")
+    rhsm_config_file = _tempfile("[nonsense]\ntest=test")
+    metrics_client = MetricsHTTPClient(config_file=config_file.name, rhsm_config_file=rhsm_config_file.name)
+    assert metrics_client.auth == ("testuser", "testpass")
+    assert not metrics_client.cert
+    assert metrics_client.base_url == "cloud.redhat.com"
+    assert metrics_client.api_prefix == "/api"
+
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_default_cert_auth(proxy_settings, config_file_factory):
+    '''
+    Verify that when insights-client configuration is used, CERT auth is selected
+    if no authmethod is defined in the configuration file
+    '''
+    config_file = config_file_factory("username=testuser\npassword=testpass")
+    rhsm_config_file = _tempfile("[nonsense]\nauth=test")
+    metrics_client = MetricsHTTPClient(
+        config_file=config_file.name, rhsm_config_file=rhsm_config_file.name,
+        cert_file="/path/to/test/cert.pem", key_file="/path/to/test/key.pem")
+    assert not metrics_client.auth
+    assert metrics_client.cert == ("/path/to/test/cert.pem", "/path/to/test/key.pem")
+    assert metrics_client.base_url == "cert.cloud.redhat.com"
+    assert metrics_client.api_prefix == "/api"
+
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_is_not_satellite(proxy_settings, config_file_factory, rhsm_config_file_factory):
+    '''
+    Verify that when the configured RHSM hostname matches one of the Red Hat subscription URLs,
+    it is determined to NOT be a Satellite-subscribed host
+    '''
+    config_file = config_file_factory("")
+    rhsm_config_file = rhsm_config_file_factory(hostname="subscription.rhsm.redhat.com")
+    metrics_client = MetricsHTTPClient(
+        config_file=config_file.name, rhsm_config_file=rhsm_config_file.name,
+        cert_file="/path/to/test/cert.pem", key_file="/path/to/test/key.pem")
+    assert not metrics_client.auth
+    assert metrics_client.cert == ("/path/to/test/cert.pem", "/path/to/test/key.pem")
+    assert metrics_client.base_url == "cert.cloud.redhat.com"
+    assert metrics_client.api_prefix == "/api"
+
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_is_not_satellite(proxy_settings, config_file_factory, rhsm_config_file_factory):
```

Comment:
❗️ This is a duplicate function name, causing the previous one to be overwritten.

```suggestion
def test_http_client_init_is_satellite(proxy_settings, config_file_factory, rhsm_config_file_factory):
```